crates/ark/src/lsp/completions/sources/composite/search_path.rs

```diff
@@ -41,13 +42,23 @@ pub(super) fn completions_from_search_path(
         let mut envir = R_GlobalEnv;
 
         while envir != R_EmptyEnv {
-            // Get environment name
-            let name = r_envir_name(envir)?;
+            let is_pkg_env = r_env_is_pkg_env(envir);
+
+            // Get package environment name, if there is one
+            let name = if is_pkg_env {
+                let name = RObject::from(r_pkg_env_name(envir));
+                let name = String::try_from(name)?;
+                Some(name)
+            } else {
+                None
+            };
```

Comment:
There was also a bug here where `r_envir_name()` was returning things like `"0x123456"` for non-package environments, which was getting fed all the way through to `.ps.help.getHtmlHelpContents()` as the package name to use.

I've changed this to restrict it to only get a `name` for a package environment

Reply:
No need to do now, but it would be useful to have an `Environment::kind()` method. It'd return an enum with variants for packages, namespaces, imports, etc, and the variant would contain the name.